src/cryptoadvance/specter/wallet.py

```diff
@@ -624,11 +626,12 @@ def balance_on_address(self, address):
         return sum(balancelist)
 
     def utxo_on_label(self, label):
-        utxo = [tx for tx in self.utxo if self.getlabel(tx["address"]) == label]
+        utxo = [tx for tx in self.utxo if self.utxo_labels_list[tx["address"]] == label]
+
         return len(utxo)
 
     def balance_on_label(self, label):
-        balancelist = [utxo["amount"] for utxo in self.utxo if self.getlabel(utxo["address"]) == label]
+        balancelist = [utxo["amount"] for utxo in self.utxo if self.utxo_labels_list[utxo["address"]] == label]
         return sum(balancelist)
```

Comment:
```suggestion
        return sum(utxo["amount"] for utxo in self.utxo if self.utxo_labels_list[utxo["address"]] == label)
```
What about a one-liner here? If you don't like it then I would suggest renaming `balancelist` to `balances`.
Should also be good to remove the list and use a generator comprehension instead, so that no unused list is created.